contracts/transfer/circuits/src/dusk_contract/execute.rs

```diff
@@ -0,0 +1,1508 @@
+        let obfuscated_commitment_one = compute_value_commitment(
+            obfuscated_note_value_one,
+            obfuscated_note_blinder_one,
+        );
+
+        // Generate the commitment to the second output note, C.c(v,b)
+        let obfuscated_note_value_two = JubJubScalar::from(200 as u64);
+        let obfuscated_note_blinder_two = JubJubScalar::from(200 as u64);
+        let obfuscated_commitment_two = compute_value_commitment(
+            obfuscated_note_value_two,
+            obfuscated_note_blinder_two,
+        );
+
+        // Assign the fee
+        let fee = BlsScalar::from(200);
+
+        let mut circuit = build_execute_circuit(
+            vec![nullifier1, nullifier2],
+            vec![note_hash1, note_hash2],
+            vec![BlsScalar::from(pos_a), BlsScalar::from(pos_b)],
+            vec![
+                tree.poseidon_branch(tree_pos_1)?.unwrap(),
+                tree.poseidon_branch(tree_pos_2)?.unwrap(),
+            ],
+            vec![sk_r1, sk_r2],
+            vec![AffinePoint::from(pk_r1), AffinePoint::from(pk_r2)],
+            vec![input_commitment_one, input_commitment_two],
+            vec![input_note_value_one.into(), input_note_value_two.into()],
+            vec![input_note_blinder_one.into(), input_note_blinder_two.into()],
+            crossover_commitment,
+            crossover_commitment_value.into(),
+            crossover_commitment_blinder.into(),
+            vec![obfuscated_commitment_one, obfuscated_commitment_two],
+            vec![
+                obfuscated_note_value_one.into(),
+                obfuscated_note_value_two.into(),
+            ],
+            vec![
+                obfuscated_note_blinder_one.into(),
+                obfuscated_note_blinder_two.into(),
+            ],
+            fee,
+        );
+
+        // Generate Composer & Public Parameters
+        let pub_params =
+            PublicParameters::setup(1 << 18, &mut rand::thread_rng())?;
+        let (pk, vk, _) = circuit.compile(&pub_params)?;
+        let proof = circuit.gen_proof(&pub_params, &pk, b"Execute")?;
+
+        let mut pi = vec![];
+        circuit
+            .input_poseidon_branches
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|branch| {
+                pi.push(PublicInput::BlsScalar(-branch.root, 0));
+            });
+        circuit
+            .nullifiers
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|nullifier| {
+                pi.push(PublicInput::BlsScalar(-nullifier, 0));
+            });
+        pi.push(PublicInput::AffinePoint(crossover_commitment, 0, 0));
+        circuit
+            .obfuscated_commitment_points
+            .as_ref()
+            .unwrap()
+            .iter()
+            .for_each(|commitment_point| {
+                pi.push(PublicInput::AffinePoint(*commitment_point, 0, 0));
+            });
```

Comment:
This seems to be another candidate for reducing code repetition. You can do something like:

```rust
fn add_circuit_public_inputs(circuit: &ExecuteCircuit, pi: Vec<PublicInput>) {
    circuit.
        input_poseidon_branches
...
}
```